pkg/scheduler/framework/runtime/framework.go

```diff
@@ -346,12 +330,39 @@ func NewFramework(r Registry, profile *config.KubeSchedulerProfile, opts ...Opti
 		fillEventToPluginMap(p, options.clusterEventMap)
 	}
 
+	// initialize plugins per individual extension points
 	for _, e := range f.getExtensionPoints(profile.Plugins) {
 		if err := updatePluginList(e.slicePtr, *e.plugins, pluginsMap); err != nil {
 			return nil, err
 		}
 	}
 
+	// initialize multiPoint plugins to their expanded extension points
+	if len(profile.Plugins.MultiPoint.Enabled) > 0 {
+		if err := f.expandMultiPointPlugins(profile, pluginsMap); err != nil {
+			return nil, err
+		}
+	}
+
+	if len(f.queueSortPlugins) != 1 {
+		return nil, fmt.Errorf("one queue sort plugin required for profile with scheduler name %q", profile.SchedulerName)
+	}
+
+	var totalPriority int64
+	if len(profile.Plugins.MultiPoint.Enabled) > 0 {
+		err, multiPointWeights := getScoreWeights(f, totalPriority, profile.Plugins.MultiPoint.Enabled)
```

Comment:
I don't understand why we need to do this here, can't we just apply the default weights once after calculating the final set of score plugins?

Reply:
Ah that's a good point, yeah we can just work with the expanded Score weights. Ignore my comment above then :)

Reply:
Actually, we can't use the expanded score weights, because we don't actually expand the weights. This function does not modify the KubeSchedulerConfig. We only build `f.scorePlugins` from the config. Then, we get the weights from the config to translate it to `f.scorePluginWeight` later on. So we still need to look at both the MultiPoint weights and the individual Score weights as I mentioned in https://github.com/kubernetes/kubernetes/pull/105611/files#r750365051